consensus/accept.go

```diff
@@ -108,9 +90,23 @@ func (s *State) validHeader(b Block) (err error) {
 		return
 	}
 
-	// Check that the block is the correct size.
-	if len(encoding.Marshal(b)) > BlockSizeLimit {
-		return LargeBlockErr
+	return
+}
+
+// addBlockToTree takes a block and a parent node, and adds a child node to the
+// parent containing the block. No validation is done.
+func (s *State) addBlockToTree(b Block) (err error) {
+	parentNode := s.blockMap[b.ParentID]
+	newNode := parentNode.newChild(b)
+
+	// Add the node to the block map
+	s.blockMap[b.ID()] = newNode
+
+	if newNode.heavierThan(s.currentBlockNode()) {
+		err = s.forkBlockchain(newNode)
+		if err != nil {
+			return
+		}
 	}
```

Comment:
I'm reasonably confident that the forking code is correct, given that it is passing the new tests. It actually failed after I merged your tests, and I had to go back and fix a bug. Good thing they were merged before these changes.